internal/controller/merge.go

```diff
@@ -213,17 +213,13 @@ func routersAreCompatible(r, toMerge *frr.RouterConfig) error {
 	return nil
 }
 
-// Verifies that two neighbors are compatible for merging, assuming they belong to the same router.
+// neighborsAreCompatible verifies that two neighbors are compatible for merging.
 func neighborsAreCompatible(n1, n2 *frr.NeighborConfig) error {
-	if n1.Addr != n2.Addr {
-		return fmt.Errorf("neighbors with different addresses (%s != %s) are not compatible for merging", n1.Addr, n2.Addr)
+	if n1.ID() != n2.ID() {
+		return fmt.Errorf("bug only neighbors with same ID are compatible for merging")
```

Comment:
let's leave a `we shouldn't reach this` comment on the `if` instead of the "bug", and format the error with the different IDs similar to before.

Reply:
done